pkg/proxy/proxy.go

```diff
@@ -339,14 +337,26 @@ func uploadToBlobStorage(ctx context.Context, cacheID int) (*DockerGHAUploadCach
 			return nil, fmt.Errorf("no presigned URL found")
 		}
 
-		client, err := azblob.NewClientWithNoCredential(cacheEntry.BackendReserveResponse.AzureBlob.PreSignedURL, nil)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create Azure Blob client: %w", err)
-		}
+		for attempt := 0; attempt < maxRetries; attempt++ {
+			contentReader := bytes.NewReader(finalBuffer.Bytes())
+			req, err := http.NewRequestWithContext(ctx, http.MethodPut, cacheEntry.BackendReserveResponse.AzureBlob.PreSignedURL, contentReader)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create Azure Blob request: %w", err)
+			}
+			req.Header.Set("Content-Type", "application/octet-stream")
+			req.Header.Set("x-ms-blob-type", "BlockBlob")
 
-		_, err = client.UploadBuffer(ctx, cacheEntry.BackendReserveResponse.AzureBlob.ContainerName, cacheEntry.BackendReserveResponse.AzureBlob.BlobName, finalBuffer.Bytes(), nil)
-		if err != nil {
-			return nil, fmt.Errorf("failed to upload to Azure Blob: %w", err)
+			resp, err := http.DefaultClient.Do(req)
+			if err == nil && resp.StatusCode >= 200 && resp.StatusCode < 300 {
+				defer resp.Body.Close()
+				break
+			} else {
+				defer resp.Body.Close()
```

Comment:
Don't defer. Call directly.

Reply:
Changed for the one in if-loop. Kept defer for the else one to be consistent with S3 case.

Reply:
ack